lib/improver/weights.py

```diff
@@ -197,16 +333,17 @@ def process(self, cube, coord):
                    ' {0:s}'.format(type(cube)))
             raise ValueError(msg)
 
-        if not cube.coords(coord):
-            msg = ('The coord for this plugin must be '
-                   'an existing coordinate in the input cube.')
-            raise ValueError(msg)
+        (num_of_weights,
+         missing_weights) = WeightsUtilities.process_coord(cube, coord)
 
-        num_of_weights = len(cube.coord(coord).points)
+        weights_in = WeightsUtilities.linear_weights(num_of_weights,
+                                                     y0val=self.y0val,
+                                                     slope=self.slope,
+                                                     ynval=self.ynval)
 
-        weights = linear_weights(num_of_weights, y0val=self.y0val,
-                                 slope=self.slope,
-                                 ynval=self.ynval)
+        redist_weights = WeightsUtilities.redistribute_weights(weights_in,
+                                                               missing_weights)
+        weights = redist_weights[np.where(redist_weights != -1.0)]
```

Comment:
Did this. Had to adjust results array in two unittests in `test_WeightsUtilities.py` also.